pymilo/exceptions/deserialize_exception.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+from .pymilo_exception import PymiloException
+
+class DeSerilaizatoinErrorTypes(Enum):
+    CORRUPTED_JSON_FILE = 1
+    INVALID_MODEL = 2
+    VALID_MODEL_INVALID_INTERNAL_STRUCTURE = 3
+
+class PymiloDeserializationException(PymiloException):
+    def __init__(self, meta_data):            
+        # Call the base class constructor with the parameters it needs
+        message = "Pymilo Deserialization failed since "
+        error_type = meta_data['error_type']
+        if(error_type == DeSerilaizatoinErrorTypes.CORRUPTED_JSON_FILE):
+            message += 'the given json file is not a valid .json file.'
+        elif(error_type == DeSerilaizatoinErrorTypes.INVALID_MODEL):
+            message += 'the given model is not supported or is not a valid model.'
+        elif(error_type == DeSerilaizatoinErrorTypes.VALID_MODEL_INVALID_INTERNAL_STRUCTURE):
+            message += 'the given model has some non-standard customized internal objects or functions.'
```

Comment:
Use `message = "Pymilo Deserialization failed since {}"` and `.format` method.
You can also use a dictionary for mapping error types to error messages instead which makes it easier for further development.

Reply:
@sadrasabouri dear Sadra, thank you for your comment, cause we support older versions of python, we can't use `.format method` and I should use the concatenation.
but I will apply your suggestion for using dict.

Reply:
Are you sure?
According to [here](https://docs.python.org/3.0/whatsnew/2.6.html#pep-3101-advanced-string-formatting), the `format` method is supported for `Python 3.0` and above.

Reply:
my bad, thanks.